src/main/java/jenkins/plugins/office365connector/HttpWorker.java

```diff
@@ -33,19 +35,28 @@
  */
 public class HttpWorker implements Runnable {
 
+    private final ExecutorService executorService = Executors.newCachedThreadPool();
+
     private final PrintStream logger;
 
     private final String url;
     private final String data;
     private final int timeout;
-    private final int retries;
 
-    public HttpWorker(String url, String data, int timeout, int retries, PrintStream logger) {
+    private static final int RETRIES = 3;
```

Comment:
as this is not configured now (each time it was set to 3) then there was no reason to pass it as the parameter each time